src/coreclr/src/jit/lowerxarch.cpp

```diff
@@ -1081,6 +1092,936 @@ void Lowering::LowerHWIntrinsic(GenTreeHWIntrinsic* node)
+
+        node->ChangeOper(GT_IND);
+        node->gtOp1 = clsVarAddr;
+
+        // TODO-XARCH-CQ: We should be able to modify at least the paths that use Insert to trivially support partial
+        // vector constants. With this, we can create a constant if say 50% of the inputs are also constant and just
+        // insert the non-constant values which should still allow some gains.
+
+        return;
+    }
+    else if (argCnt == 1)
+    {
+        // We have the following:
+        //
+        // /--*  <value>
+        // *  HWINTRINSIC simd16 T Create
+
+        if (intrinsicId == NI_Vector256_Create)
+        {
+            if (comp->compOpportunisticallyDependsOn(InstructionSet_AVX2))
+            {
+                // We will be constructing the following parts:
+                //
+                //      /--*  <value>
+                // t1 = *  HWINTRINSIC simd16 ubyte CreateScalarUnsafe
+                //      /--*  t1     simd16
+                // t0 = *  HWINTRINSIC simd32 T BroadcastScalarToVector256
+
+                op1 = comp->gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_Vector128_CreateScalarUnsafe, baseType, 16);
+                BlockRange().InsertBefore(node, op1);
+                LowerNode(op1);
+
+                node->gtOp1 = op1;
+                node->gtOp2 = nullptr;
+
+                node->gtHWIntrinsicId = NI_AVX2_BroadcastScalarToVector256;
+                return;
+            }
+
+            assert(comp->compIsaSupportedDebugOnly(InstructionSet_AVX));
+
+            // We will be constructing the following parts:
+            //
+            // t2 =    CNS_INT   int    0
+            //      /--*  t?     simd32
+            //      +--*  t?     simd16
+            //      +--*  t?     int
+            // t0 = *  HWINTRINSIC simd32 T InsertVector128
+
+            op2 = comp->gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, NI_Vector128_Create, baseType, 16);
+            BlockRange().InsertBefore(node, op2);
+            LowerNode(op2);
+
+            node->gtOp2 = op2;
+
+            LIR::Use op2Use(BlockRange(), &node->gtOp2, node);
+            ReplaceWithLclVar(op2Use);
+
+            op1 = comp->gtClone(node->gtOp2);
+            BlockRange().InsertBefore(node, op1);
+            LowerNode(op1);
+
+            op1 = comp->gtNewSimdHWIntrinsicNode(TYP_SIMD32, op1, NI_Vector128_ToVector256Unsafe, baseType, 16);
+            BlockRange().InsertBefore(node, op1);
+            LowerNode(op1);
+
+            idx = comp->gtNewIconNode(0x01, TYP_INT);
+            BlockRange().InsertBefore(node, idx);
+
+            node->gtOp1 = comp->gtNewArgList(op1, node->gtOp2, idx);
+            node->gtOp2 = nullptr;
+
+            node->gtHWIntrinsicId = NI_AVX_InsertVector128;
+            return;
+        }
+
+        // We will be constructing the following parts:
+        //
+        //      /--*  <value>
+        // t1 = *  HWINTRINSIC simd16 ubyte CreateScalarUnsafe
```

Comment:
It might be nice to give 't1' a more meaningful name (since you refer to it in the comments below)

Reply:
The comments were updated to be more descriptive and give a "managed equivalent" using the same names as used in the LIR comments, which should hopefully help devs understand what is being used where and why.